channeldb/invoices.go

```diff
@@ -848,9 +872,16 @@ func (d *DB) UpdateInvoice(paymentHash lntypes.Hash,
 
 		// Check the invoice index to see if an invoice paying to this
 		// hash exists within the DB.
-		invoiceNum := invoiceIndex.Get(paymentHash[:])
+		var invoiceNum []byte
+		if payAddr != nil {
+			payAddrIndex := tx.ReadWriteBucket(payAddrIndexBucket)
```

Comment:
Does this need to be a `ReadWriteBucket`? If not, maybe code from `LookupInvoice` can be reused

Reply:
Functions that want a read bucket can also have a read-write bucket passed in. However in this case, we don't modify the bucket all, so we can indeed make due with a read bucket here.